### PR TITLE
feat: update definitions to support ABAC Conditions

### PR DESCRIPTION
**Pull request by jpadilla**
## Description
NOTE: this PR targets a new branch `feat/abac`

- ListObjectsRequest now accepts `context`
- StreamedListObjectsRequest now accepts `context`
- WriteAuthorizationModelRequest now accepts `conditions`
- CheckRequest now accepts `context`
- ExpandRequest stops accepting `user`

## Review Checklist
- [ ] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [ ] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: docs/openapiv2/apidocs.swagger.json [hunk 1]
```diff
@@ -1107,6 +1131,42 @@
         }
       }
     },
+    "Condition": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "title": "A unique name for the condition\nTODO(@jpadilla): Add validation rules"
```
Comment (miparnisari) [inline docs/openapiv2/apidocs.swagger.json:1139]: remove this TODO from here

File: openfga/v1/authzmodel.proto [hunk 2]
```diff
@@ -126,3 +131,25 @@ message TupleToUserset {
   ObjectRelation tupleset = 1;
   ObjectRelation computed_userset = 2;
 }
+
+message Condition {
+  // A unique name for the condition
+  // TODO(@jpadilla): Add validation rules
```
Comment (miparnisari) [inline openfga/v1/authzmodel.proto:137]: In this PR?

**Review by miparnisari — CHANGES_REQUESTED** [review]

File: docs/openapiv2/apidocs.swagger.json (hunk 1 above)
Comment (jpadilla) [reply docs/openapiv2/apidocs.swagger.json:1139]: let's follow up on a separate pull request.

File: openfga/v1/authzmodel.proto (hunk 2 above)
Comment (jpadilla) [reply openfga/v1/authzmodel.proto:137]: let's follow up on a separate pull request.

File: docs/openapiv2/apidocs.swagger.json (hunk 1 above)
Comment (jpadilla) [reply docs/openapiv2/apidocs.swagger.json:1139]: this gets picked up automatically, i'll remove the actual TODO for now from this pull request

File: openfga/v1/authzmodel.proto (hunk 2 above)
Comment (jon-whit) [reply openfga/v1/authzmodel.proto]: Validation rules are part of the API spec though. Splitting that to another PR is just going to create more noise and context switching. Please include it here in the scope of this PR.

**Comment by rhamzeh** [comment]
Converted to draft so that we don't accidentally merge

File: openfga/v1/authzmodel.proto [hunk 3]
```diff
@@ -126,3 +139,55 @@ message TupleToUserset {
...
+      min_len: 1,
+      max_len: 50,
+    }
+  ];
+}
+
+message ConditionParamTypeRef {
+  enum TypeName {
+    TYPE_NAME_UNSPECIFIED = 0;
+    TYPE_NAME_ANY = 1;
```
Comment (rhamzeh) [inline openfga/v1/authzmodel.proto:174]: We're keeping any?

File: openfga/v1/authzmodel.proto [hunk 4]
```diff
@@ -21,6 +21,16 @@ message AuthorizationModel {
...
     (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {example: "[{\"type\": \"user\"}, {\"type\":\"document\",\"relations\":{\"reader\":{\"union\":{\"child\":[{\"this\":{}},{\"computedUserset\":{\"object\":\"\",\"relation\":\"writer\"}}]}},\"writer\":{\"this\":{}}},\"metadata\":{\"relations\":{\"reader\":{\"directly_related_user_types\":[{\"type\":\"user\"}]},\"writer\":{\"directly_related_user_types\":[{\"type\":\"user\"}]}}}}]"}
   ];
+
+  map<string, Condition> conditions = 4 [
+    json_name = "conditions",
+    (validate.rules).map.max_pairs = 25,
+    (validate.rules).map.keys.string = {
+      pattern: "^[^:#@\\s]{1,50}$",
+      min_len: 1,
+      max_len: 50,
```
Comment (miparnisari) [inline openfga/v1/authzmodel.proto:31]: why do we need this? seems redundant

File: openfga/v1/authzmodel.proto (hunk 3 above)
Comment (miparnisari) [reply openfga/v1/authzmodel.proto:174]: If in the future we'll introduce it, it makes sense to have it here so that we reserve the field number

**Review by miparnisari — APPROVED** [review]

**Pull request merged**